tests/runtime_test/runtime_test.c

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2022, Intel Corporation */
+
+#include <libminiasync.h>
+#include <libminiasync-vdm-dml.h>
+#include <stdlib.h>
+#include <time.h>
+#include <string.h>
+#include "util_dml.h"
+#include "core/os.h"
+#include "test_helpers.h"
+
+/*
+ * Test executing multiple futures with different data movers
+ */
+int
+test_runtime_wait_multiple(size_t str_len)
```

Comment:
You need to add test witch custom features (not using vdm one),

Test the following cases: 
- Chain future with has async and non async futures.
- Future which changes its flag during execution.